pkg/params/mergeparams.go

```diff
@@ -121,7 +121,12 @@ func ParseParams(params []string) (map[string]string, error) {
 		if len(r) != 2 {
 			return nil, errors.New(invalidParam + p)
 		}
-		parsedParams[r[0]] = r[1]
+		key := strings.TrimSpace(r[0])
+		val := strings.TrimSpace(r[1])
+		if key == "" {
+			return nil, errors.New(invalidParam + p)
+		}
```

Comment:
@piyush-garg I don't recall why we didn't add this check at the first place, are having a case where key can be `""` 🤔

Reply:
I can't think of any usefulness of an empty key, JSON does allow for it. I thought being a bit more strict here is better at catching typos...

Reply:
I think it may be missed, but I don't think empty key can be of any use.